lib/worker/src/routes/vaults.rs

```diff
@@ -0,0 +1,200 @@
+            &GetObjectRequest {
+                bucket: gcs_client.bucket.to_string(),
+                object: cache_path.unwrap().as_ref().to_string(),
+                ..Default::default()
+            },
+            &Range::default(),
+        )
+        .await
+    {
+        Ok(s) => s,
+        Err(err) => {
+            log::error!("{}", err);
+            return HttpResponse::InternalServerError().json(format!(
+                "{{\"error\": \"{}\"}}",
+                "failed to download record"
+            ));
+        }
+    };
+
+    HttpResponse::Ok()
+        .content_type(ContentType::octet_stream())
+        .streaming(stream)
+}
+
+impl TryFrom<web::Path<std::string::String>> for Cid {
+    type Error = String;
+
+    fn try_from(value: web::Path<std::string::String>) -> Result<Cid, String> {
+        Cid::from(value.to_string())
+    }
+}
+
+pub async fn find_vaults_by_account<E: EVMClient + 'static + std::marker::Sync>(
+    params: web::Query<FindVaultsByAccountParams>,
+    evm_client: web::Data<E>,
+) -> HttpResponse {
+    let account = match Address::from_str(params.account.as_str()) {
+        Ok(v) => v,
+        Err(_) => {
+            return HttpResponse::BadRequest()
+                .json(format!("{{\"error\": \"{}\"}}", "account is invalid"));
+        }
+    };
+
+    let vaults = match evm_client.list_pub(account).await {
+        Ok(vaults) => vaults,
+        Err(err) => {
+            return HttpResponse::BadRequest().json(format!("{{\"error\": \"{}\"}}", err));
+        }
+    };
+
+    HttpResponse::Ok().json(vaults)
+}
+
+#[derive(Debug, Deserialize)]
+pub struct FindVaultsByAccountParams {
+    account: String,
+}
+
+pub async fn find_records_by_pub_id(
+    path: web::Path<String>,
+    params: web::Query<FindRecordsByPubIdQueryParams>,
+    pool: web::Data<PgPool>,
+) -> HttpResponse {
+    let vault: Vault = match path.try_into() {
+        Ok(p) => p,
+        Err(err) => return HttpResponse::BadRequest().json(format!("{{\"error\": \"{}\"}}", err)),
+    };
+
+    let records = match db::pub_cids(
+        &pool,
+        &vault,
+        params.limit(),
+        params.offset(),
+        params.before(),
+        params.after(),
+    )
+    .await
+    {
+        Ok(records) => records,
+        Err(_) => {
+            return HttpResponse::InternalServerError()
+                .json(format!("{{\"error\": \"{}\"}}", "Failed to fetch deals"))
+        }
+    };
+
+    HttpResponse::Ok().json(records)
+}
+
+impl TryFrom<web::Path<std::string::String>> for Vault {
+    type Error = String;
+
+    fn try_from(value: web::Path<std::string::String>) -> Result<Vault, String> {
+        Vault::from(value.to_string())
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct FindRecordsByPubIdQueryParams {
+    limit: Option<i32>,
+    offset: Option<i32>,
+    before: Option<String>,
+    after: Option<String>,
+}
+
+impl FindRecordsByPubIdQueryParams {
+    fn limit(&self) -> i32 {
```

Comment:
thinking if we could deprecate `limit` and `offset`?  `before` and `after` serves the same purpose 🤔

Reply:
actually, they don't serve the same purpose. `limit` and `offset` are used for pagination and `before` and `after` filtering